server/api/login.go

```diff
@@ -145,6 +145,41 @@ func HandleAuth(c *gin.Context) {
 		return
 	}
 
+	repos, err := server.Config.Services.Forge.Repos(c, u)
+	if err != nil && !errors.Is(err, types.RecordNotExist) {
+		log.Error().Msgf("cannot list repos for %s. %s", u.Login, err)
+		c.Redirect(http.StatusSeeOther, "/login?error=internal_error")
+		return
+	}
+
+	for _, forgeRepo := range repos {
+		dbRepo, err := _store.GetRepoForgeID(forgeRepo.ForgeRemoteID)
+		if err != nil && errors.Is(err, types.RecordNotExist) {
+			continue
+		}
+		if err != nil {
+			log.Error().Msgf("cannot list repos for %s. %s", u.Login, err)
+			c.Redirect(http.StatusSeeOther, "/login?error=internal_error")
```

Comment:
i guess this relates to line 148 and following, the db query here should always succeed.
i really don't know about swallowing _all_ errors. you would want to let users know that it can't list the repos from the forge, wouldn't you?
So _imho_ if one wanted to ignore rate-limit errors here, it should check only for that. (which would require re-working _all_ the forge clients)

Reply:
that's completely besides what i saying but ok...